acorn/src/expression.js

```diff
@@ -167,6 +167,11 @@ pp.parseMaybeConditional = function(forInit, refDestructuringErrors) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprOps(forInit, refDestructuringErrors)
   if (this.checkExpressionErrors(refDestructuringErrors)) return expr
+
+  return this.parseConditional(expr, startPos, startLoc, forInit, refDestructuringErrors)
+}
+
+pp.parseConditional = function(expr, startPos, startLoc, forInit, refDestructuringErrors) {
```

Comment:
This only parses a conditional when it sees a `?` token. Either move that check out of the function into `parseMaybeConditional`, or, if that doesn't help in your code, leave this as it was and duplicate the entire method in the plugin.